### PR TITLE
Add resolver for searching both MapLayers and MapCollections

### PR DESCRIPTION
**Pull request by jonkoops**

### CODE REVIEW

File: src/graphql/graphql.schema.ts [hunk 1]
```diff
@@ -3,23 +3,25 @@ const gql = (input: any) => input
...
   interface SearchResult {
     totalCount: Int!
+    # TODO: See if results can be made required.
     results: [Results!]
     filters: [Filter!]
     pageInfo: PageInfo!
   }
 
-  interface SearchResultType {
+  interface MixedResult {
```
Comment (royderks) [inline src/graphql/graphql.schema.ts:21]: Not sure about using `Mixed*` as naming for typings, can we rename this to something more understandable?

File: src/graphql/search/data/index.ts [hunk 2]
```diff
@@ -55,7 +51,7 @@ const index = async (
   }
 
   // Get the results from the DataLoader
-  const dataloaderResults: DataSearchResultType[] = await Promise.all(
+  const dataloaderResults: MixedDataResult[] = await Promise.all(
```
Comment (royderks) [inline src/graphql/search/data/index.ts:54]: Same, not sure to expect when reading mixed data result

File: src/graphql/graphql.schema.ts [hunk 3]
```diff
@@ -92,11 +95,19 @@ const schema = gql`
...
+    pageInfo: PageInfo!
+  }
+
   type Filter {
     type: String!
     label: String!
-    options: [FilterOptions!]
-    filterType: String
+    options: [FilterOptions!]!
+    # TODO: Make filterType a proper enum.
```
Comment (royderks) [inline src/graphql/graphql.schema.ts:109]: Do we need a new issue for this?

Comment (royderks) [reply src/graphql/graphql.schema.ts:109]: Or can you use the already exisiting `FilterType` for this?

Comment (jonkoops) [reply src/graphql/graphql.schema.ts:109]: I added the enum to the schema, and replaced the existing FilterType instances

**Pull request merged**